app/src/main/res/layout/main_activity.xml

```diff
@@ -25,26 +25,14 @@
         android:layout_height="match_parent"
         tools:context=".MainActivity">
 
-        <com.google.android.material.navigationrail.NavigationRailView
-            android:id="@+id/navigation_rail"
-            android:layout_width="wrap_content"
-            android:layout_height="0dp"
-            app:layout_constraintBottom_toBottomOf="parent"
-            app:layout_constraintStart_toStartOf="parent"
-            app:layout_constraintTop_toTopOf="parent"
-            app:headerLayout="@layout/navigation_rail_header"
-            app:isGone="@{!@bool/show_navigation_rail}"
-            app:labelVisibilityMode="unlabeled"
-            app:menu="@menu/navigation_rail" />
-
         <androidx.fragment.app.FragmentContainerView
             android:id="@+id/nav_host_fragment"
             android:name="androidx.navigation.fragment.NavHostFragment"
-            android:layout_width="0dp"
-            android:layout_height="0dp"
+            android:layout_width="match_parent"
```

Comment:
Hmm, i had thought `ConstraintLayout` prefers using `0dp` over `match_parent` when there are constraints on that dimension. (At least, I recall getting lint warnings to that effect once upon a time.) If that's not the case though, then please disregard.
Same for line 32.

Reply:
Yep, you're right. 0dp within ConstraintLayout is the equivalent of MATCH_CONSTRAINT: https://developer.android.com/reference/androidx/constraintlayout/widget/ConstraintLayout#widgets-dimension-constraints

This doesn't work well in layout tools, though, so I'll change this to 0dp but add tools:layout_height and tools_layout_width attributes,

Reply:
When I set this to 0dp, the content all disappears. match_parent is providing the expected behavior in all cases, though. 

Any thoughts on how to better debug this? I tried layout inspector and the right layouts are there, the width is being set to 0 though.